homeTask2/src/hooks/useUsers.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import { UserItem } from "../components/types";
+
+const useUsers = () => {
+    const [userHistoryList, setUserHistoryList] = useState(JSON.parse(localStorage.getItem("UserHistory") || "[]"));
+    const addUser = (event: any, userInfo: UserItem) => {
+        event.preventDefault();
+        const timestamp = Date.now();
+        const user = Object.assign({}, userInfo);
+        user.id = timestamp;
+        user.created = timestamp.toLocaleString();
+        const newUsers = [...userHistoryList, user];
+        setUserHistoryList(newUsers);
+        localStorage.setItem("UserHistory", JSON.stringify(newUsers));
+    };
+
+    const editUser = (id: number, userInfo: UserItem) => {
+        const newUserHistory = userHistoryList.map(item => {
+            if (item.id.toString() === id) {
+                item.firstName = userInfo.firstName;
+                item.lastName = userInfo.lastName;
+                item.userName = userInfo.userName;
+                item.language = userInfo.language;
+                return item
```

Comment:
```suggestion
                return { ...item, ...userInfo }
```
this will return a new object with values of `item` updated with values of `useInfo`